### PR TITLE
[sdl2] Prevent exports in static build

### PR DESCRIPTION
**Pull request by codicodi**
Fixes https://github.com/Microsoft/vcpkg/issues/1351.
Needed to bump a few dependent libraries, as they all depend on SDL's `DECLSPEC` macro.

### CODE REVIEW

File: ports/sdl2-image/CMakeLists.txt [hunk 1]
```diff
@@ -27,24 +27,26 @@ set(WEBP_DYNAMIC \"libwebp-4.dll\")
...
+    IMG_tga.c
+    IMG_tif.c
+    IMG_webp.c
+    IMG_xcf.c
+    IMG_xpm.c
+    IMG_xv.c
+    IMG_xxx.c
+    )
+
+set_target_properties(SDL2_image PROPERTIES DEFINE_SYMBOL SDL2_EXPORTS)
```
Comment (ras0219-msft) [inline ports/sdl2-image/CMakeLists.txt]: Can we handle this through post-processing the headers?

Comment (codicodi) [inline ports/sdl2-image/CMakeLists.txt]: You mean to modify headers installed by sdl2? Not really, those exports are for sdl2-image itself (as I pointed before it uses the same `DECLSPEC ` macro to tag its symbols and therefore needs to share the same `SDL2_EXPORTS` flag). It's kinda messy, but should work fine in all sensible combinations now.

Comment (ras0219-msft) [inline ports/sdl2-image/CMakeLists.txt]: I see, fair enough.

Comment (ras0219-msft) [inline ports/sdl2-image/CMakeLists.txt]: Hang on -- Shouldn't this macro only be set for `BUILD_SHARED_LIBS=1`?

Comment (codicodi) [inline ports/sdl2-image/CMakeLists.txt]: `DEFINE_SYMBOL` is intended just for this purpose and is ignored in static builds: 

> DEFINE_SYMBOL sets the name of the preprocessor symbol defined when compiling sources in a **shared library**

**Comment by ras0219-msft** [comment]
Awesome job, and thanks for the explanation (TIL)!

**Pull request merged**